drivers/adc/adc_stm32.c

```diff
@@ -801,6 +796,100 @@ static void adc_stm32_setup_speed(const struct device *dev, uint8_t id,
 #endif
 }
 
+static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_id)
```

Comment:
While you are at it, could you check if it is necessary to disable the ADC here before configuring the common path?
Some series requires the ADC to be disabled, see [this](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32g0xx/drivers/include/stm32g0xx_ll_adc.h#L1810)

Also, do you think if the common path should be remove after sampling?

Reply:
Also:

> I guess this [line](https://github.com/zephyrproject-rtos/zephyr/blob/5ee306944b37fac3143f30e759478c6e8a9d36a4/drivers/adc/adc_stm32.c#L987) is not required in the init anymore, since we have [this](https://github.com/zephyrproject-rtos/zephyr/blob/5ee306944b37fac3143f30e759478c6e8a9d36a4/drivers/adc/adc_stm32.c#L852)?

Reply:
> While you are at it, could you check if it is necessary to disable the ADC here before configuring the common path?
Some series requires the ADC to be disabled, see [this](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32g0xx/drivers/include/stm32g0xx_ll_adc.h#L1810)

oof this could be problematic. Configuring one ADC while another is preforming an on-going calculation could get messy. Perhaps we would just wait for another ADC conversion run to finish before configuring the internal channel on the models? @FRASTM @erwango ?

Reply:
@pdietl Indeed, let's keep is simple and safe for now.